azure/packages/azure-client/src/test/AzureClient.spec.ts

```diff
@@ -45,239 +45,614 @@ describe("AzureClient", () => {
+
+            await new Promise<void>((resolve) => {
+                container.on("connected", () => {
+                    resolve();
+                });
+            });
+
+            const resources = client.copyContainer(containerId, schema);
+            await assert.doesNotReject(resources, () => true, "container could not be copied");
+
+            const { container: containerCopy } = await resources;
+
+            const map1Get = containerCopy.initialObjects.map1 as SharedMap;
+            const valueGet: string | undefined = await mapWait(map1Get, "new-key");
+            assert.strictEqual(valueGet, valueCreate, "DDS value was not correctly copied.");
+        });
+
+        /**
+         * Scenario: test if Azure Client can handle non-existing container when trying to copy
+         *
+         * Expected behavior: client should throw an error.
+         */
+        it("can handle non-existing container", async () => {
+            const resources = client.copyContainer("badidoncopy", schema);
+            await assert.rejects(resources, () => true, "We should not be able to copy container.");
+        });
     });
 
-    /**
-     * Scenario: test if Azure Client can get recreate container.
-     *
-     * Expected behavior: an error should not be thrown nor should a rejected promise
-     * be returned.
-     */
-    /* PR #9650 Needs to be merged for full flow to work
-    it("can copy old document successfully", async () => {
-        const newContainer = (await client.createContainer(schema)).container;
-        // const containerId = await newContainer.attach();
-        await new Promise<void>((resolve) => {
-            newContainer.on("connected", () => {
-                resolve();
+    describe("Fluid audience", () => {
+        /**
+         * Scenario: Find original member/self
+         *
+         * Expected behavior: container should have a single member upon creation.
+         */
+        it("can find original member", async () => {
+            const { container, services } = await client.createContainer(schema);
+            const containerId = await container.attach();
+            await new Promise<void>((resolve) => {
+                container.on("connected", () => {
+                    resolve();
+                });
             });
+
+            assert.strictEqual(typeof containerId, "string", "Attach did not return a string ID");
+            assert.strictEqual(
+                container.attachState,
+                AttachState.Attached,
+                "Container is not attached after attach is called",
+            );
+
+            /* This is a workaround for a known bug, we should have one member (self) upon container connection */
+            if (services.audience.getMembers().size === 0) {
+                await new Promise<void>((resolve) => {
+                    services.audience.on("memberAdded", () => {
+                        resolve();
```

Comment:
This is sufficient here, but might be good to go ahead and add the check to make sure the added member is myself, in case this gets copied and used elsewhere later.  Maybe even pull out into a helper function similar to mapWait, like a waitForMyself(audience)
```suggestion
                        if (services.audience.getMyself() !== undefined) {
                            resolve();
                        }
```

Reply:
That's a great point. pulled out a helper that's waiting for self.